astropy/io/fits/hdu/compressed.py

```diff
@@ -882,10 +882,14 @@ def _update_header_data(self, image_header,
             if bzero != 0.0:
                 self._header.set('BZERO', bzero, after=after_keyword)
 
+        try:
             bitpix_comment = image_header.comments['BITPIX']
-            naxis_comment = image_header.comments['NAXIS']
-        else:
+        except (AttributeError, KeyError):
```

Comment:
tl;dr -- Does this violate the FITS standards?

Reply:
Do you mean does it violate if the header doesn't have `BITPIX` and/or `NAXIS`? This fix is intended to allow the `header` passed to `CompImageHDU` during initialisation to only contain *extra* cards that should be appended to the header that is automatically generated if `header=None`. So the header of the initialised HDU would still comply with the FITS standards.

Reply:
I dunno. I never had enough willpower to read the standards. I just want to put on records here that this point has been addressed. 😉